src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -89,30 +85,72 @@ public static void AddTodoTemplateSwaggerGen(this IServiceCollection services)
     {
         services.AddSwaggerGen(options =>
         {
-            options.AddSecurityDefinition("bearerAuth", new OpenApiSecurityScheme
-            {
-                Name = "Authorization",
-                Type = SecuritySchemeType.Http,
-                Scheme = "bearer",
-                BearerFormat = "JWT",
-                In = ParameterLocation.Header,
-                Description = "JWT Authorization header using the Bearer scheme."
-            });
+            options.AddSecurityDefinition("bearerAuth",
+                new OpenApiSecurityScheme
+                {
+                    Name = "Authorization",
+                    Type = SecuritySchemeType.Http,
+                    Scheme = "bearer",
+                    BearerFormat = "JWT",
+                    In = ParameterLocation.Header,
+                    Description = "JWT Authorization header using the Bearer scheme."
+                });
 
             options.AddSecurityRequirement(new OpenApiSecurityRequirement
             {
                 {
                     new OpenApiSecurityScheme
                     {
-                        Reference = new OpenApiReference
-                        {
-                            Type = ReferenceType.SecurityScheme,
-                            Id = "bearerAuth"
-                        }
+                        Reference = new OpenApiReference { Type = ReferenceType.SecurityScheme, Id = "bearerAuth" }
                     },
-                    new string[] {}
+                    new string[] { }
                 }
             });
         });
     }
+
+    public static void AddHealthChecks(
+        this IServiceCollection services,
+        IConfiguration configuration,
+        IWebHostEnvironment environment)
+    {
+        var appsettings = configuration.GetSection(nameof(AppSettings)).Get<AppSettings>();
+
+        var healthCheckSetting = appsettings.HealCheckSettings;
+        
+        if (!healthCheckSetting.EnableHealthChecks)
+            return;
+
+        var emailSettings = appsettings.EmailSettings;
+
+        services
+            .AddHealthChecksUI(setupSettings: setup =>
+            {
+                setup.AddHealthCheckEndpoint("BitHealthCheck",
+                    environment.IsDevelopment() ? "http://localhost:5000/healthz" : "/healthz");
+            }).AddInMemoryStorage();
+
+
+        services.AddHealthChecks()
+            .AddProcessAllocatedMemoryHealthCheck(healthCheckSetting.MaximumMegabytesAllocated)
+            .AddDiskStorageHealthCheck(opt =>
+                opt.AddDrive(Path.GetPathRoot(Directory.GetCurrentDirectory()), minimumFreeMegabytes: 5 * 1024))
```

Comment:
isn't it better to put the hard-coded value of `5 * 1024` into the settings?
@ysmoradi

Reply:
I thinks it's fine to use hard coded values
But the value itself is too small
I'd recommend 6 * 1024 * 1024

Reply:
hmmm, since there is a setting introduced to store `MaximumMegabytesAllocated` in this PR, we need to decide which way is better, using the hard-coded values or using the settings. I'm ok with either way, but not the current mix & match 😁
@ysmoradi

Reply:
I think it better to let the end-user decide about their application memory usage